codecs/json-codec/src/main/java/software/amazon/smithy/java/json/JsonSettings.java

```diff
@@ -115,6 +118,15 @@ public boolean serializeTypeInDocuments() {
         return serializeTypeInDocuments;
     }
 
+    /**
+     * The error type sanitizer to use for {@code __type}. Default is null
```

Comment:
Hm, __type can be used for more than just error discrimination though. Maybe this should be more just a typeSanitizer.